tests/dragonfly/replication_test.py

```diff
@@ -2411,3 +2411,21 @@ async def test_replicate_old_master(
     assert await c_replica.execute_command("get", "k1") == "v1"
 
     await disconnect_clients(c_master, c_replica)
+
+
+@pytest.mark.asyncio
+async def test_empty_hash_as_zipmap_bug(async_client):
+    await async_client.execute_command("HSET foo a_field a_value")
+    await async_client.execute_command("HSETEX foo 1 b_field b_value")
+    await async_client.execute_command("HDEL foo a_field")
+
+    await asyncio.sleep(2)
+
+    @assert_eventually
+    async def check_if_empty():
+        assert await async_client.execute_command("HGETALL foo") == []
+
+    await check_if_empty()
+
+    # Key does not exist and it's empty
+    assert await async_client.execute_command(f"EXISTS foo") == 0
```

Comment:
This clearly doesn't belong to replication test 🙂

Reply:
Originally, I had the exact same chore as you did (see previous commits) however with the recent changes, we won't even reach that stage. I can move it, any particular place you want ?

Reply:
I assume this belongs to the previous comment. But why can't we have instances already running that have empty listpacks

Reply:
Not sure, maybe generic test or anything

Reply:
maybe start a hash_test

Reply:
@dranikpg  @kostasrim  why it's not a unit test?